openapi-cli/src/main/java/io/ballerina/openapi/generators/client/FunctionBodyGenerator.java

```diff
@@ -200,31 +210,40 @@ private void handleParameterSchemaInOperation(Map.Entry<PathItem.HttpMethod, Ope
         }
 
         if (operation.getValue().getParameters() != null) {
-            List<Parameter> parameters = operation.getValue().getParameters();
-            List<Parameter> queryParameters = new ArrayList<>();
-            List<Parameter> headerParameters = new ArrayList<>();
-            for (Parameter parameter: parameters) {
-                if (parameter.getIn().trim().equals("query")) {
-                    queryParameters.add(parameter);
-                } else if (parameter.getIn().trim().equals("header")) {
-                    headerParameters.add(parameter);
-                }
+            handleOperationWithParameters(operation, statementsList, queryApiKeyNameList, headerApiKeyNameList);
+        } else {
+            handleOperationWithoutParameters(statementsList, queryApiKeyNameList, headerApiKeyNameList);
+        }
+    }
+
+    /**
+     * Handle query parameters and headers when parameters are present in remote function signature.
+     */
+    private void handleOperationWithParameters(Map.Entry<PathItem.HttpMethod, Operation> operation, List<StatementNode>
+            statementsList, List<String> queryApiKeyNameList, List<String>
+                                                       headerApiKeyNameList) throws BallerinaOpenApiException {
+        List<Parameter> parameters = operation.getValue().getParameters();
+        List<Parameter> queryParameters = new ArrayList<>();
+        List<Parameter> headerParameters = new ArrayList<>();
+        for (Parameter parameter: parameters) {
+            if (parameter.getIn().trim().equals("query")) {
```

Comment:
Shall we use the [existing constant](https://github.com/ballerina-platform/openapi-tools/blob/eff4ba6a2e7c23fc2d99b0846ea5d47576366c64/openapi-cli/src/main/java/io/ballerina/openapi/generators/GeneratorConstants.java#L197) for "query", and create a constant for "header" as well?

Reply:
Done